api/services/account_service.py

```diff
@@ -184,6 +184,22 @@ def create_tenant_member(tenant: Tenant, account: Account, role: str = 'normal')
         db.session.commit()
         return ta
 
+    @staticmethod
+    def rename_tenant(account: Account, tenant_id: str, name: str) -> Tenant:
+        """Rename tenant"""
+
+        ta = TenantAccountJoin.query.filter_by(tenant_id=tenant_id, account_id=account.id).first()
+        if ta is None or ta.role == TenantAccountJoinRole.NORMAL.value:
+            TenantNotFound("Tenant not found for the account.")
+        if ta.role == TenantAccountJoinRole.NORMAL.value:
+            raise NoPermissionError("You have no permission to rename the tenant.")
```

Comment:
We replaced the word "tenant" with "workspace" earlier. 🏢 Although there are still many "tenant" that haven't been changed yet.